src/extension/ScriptRunner.ts

```diff
@@ -173,4 +186,127 @@ export class ScriptRunner {
+          const content = decoder.decode(data)
+          const resource_error = this.getResourceError(content)
+
+          const pathRegex = /((\\|\/)[a-z0-9\s_@\-^!#$%&+=.{}]+)+/i
+          const locationRegex = /\d+:\d+/g
+
+          resource_error.topics.forEach(topic => {
+            if (topic.messages.length > 0) {
+              topic.messages.forEach(message => {
+                const curMessage: string = message.message
+
+                const path = pathRegex.exec(curMessage)
+                const location = locationRegex.exec(curMessage)
+                const problemPath = this.getPath(path, confFile)
+
+                const descriptiveMessage = (topic.name + ':' + curMessage)
+                  .replace(pathRegex, '')
+                  .replace(locationRegex, '')
+                  .replace(/:{2,}/g, ':')
+
+                const position: Position = this.getPosition(
+                  location,
+                  path,
+                  confFile,
+                )
+                const range: Range = new Range(position, position)
+                const diagnostic = new Diagnostic(range, descriptiveMessage)
+
+                const diagnostics: Diagnostic[] = []
+                diagnostics.push(diagnostic)
+
+                const uri = Uri.parse(problemPath)
+
+                const curDiagnosticCollection: vscode.DiagnosticCollection =
+                  vscode.languages.createDiagnosticCollection()
+
+                curDiagnosticCollection.set(uri, diagnostics)
+                this.diagnosticCollection.push(curDiagnosticCollection)
+              })
+            }
+          })
+        })
+      })
+  }
+
+  private getResourceError(str: string): ResourceError {
+    const jsonContent: ResourceError = JSON.parse(str)
+
+    const resource_error: ResourceError = {
+      topics: [],
+    }
+
+    jsonContent.topics.forEach(topic => {
+      const curTopic: Topic = {
+        name: topic.name,
+        messages: [],
+      }
+      const curMessages: Message[] = []
+      topic.messages.forEach(message => {
+        const curMessage: Message = {
+          message: message.message,
+          location: message.location,
+        }
+        curMessages.push(curMessage)
+      })
+      curTopic.messages = curMessages
+      resource_error.topics.push(curTopic)
+    })
+
+    return resource_error
+  }
+
+  private getPosition(
+    location: RegExpExecArray | null,
+    path: RegExpExecArray | null,
+    confFilePath: string,
+  ): Position {
+    // TODO: if the path is to the configuration file, the position should be 0,0 anyway
+
+    if (path && !path.includes(confFilePath) && location) {
+      const rowPosition = parseInt(location[0].split(':')[0]) - 1
+      const colPosition = parseInt(location[0].split(':')[1]) - 1
+      return new Position(rowPosition, colPosition)
+    }
+    return new Position(0, 0)
+  }
+
+  private getPath(path: RegExpExecArray | null, confFile: string): string {
```

Comment:
Nit-picking - can you rename the function to be more explicit - which path are we trying to get? Alternatively, add a short docstring right above this function.

Reply:
It's not nit-picking if it bothers you :) I changed the function name and added a docstring with a short documentation.